app/containers/Table/index.js

```diff
@@ -88,6 +89,39 @@ const SpinnerWrapper = styled.div`
   transform: translate(-50%, -50%)
 `;
 
+function isRebuyNeeded(props, nextProps) {
+  const prevHand = nextProps.prevHand && nextProps.prevHand.toJS();
+  const hand = nextProps.hand && nextProps.hand.toJS();
+  const bb = nextProps.data ? nextProps.data.get('smallBlind') * 2 : 0;
+
+  if (!prevHand || !hand) {
+    return false;
+  }
+
+  const { myStack, standingUp, state } = nextProps;
+  const { myStack: prevStack, state: prevState } = props;
+
+  const rebuyModal = storageService.getItem(`rebuyModal[${props.params.tableAddr}${nextProps.latestHand}1]`);
+
+  if (prevHand) {
+    return !!(
+      hand.state === 'waiting' &&
+      myStack !== null && (myStack <= 0 || myStack < bb) &&
+      !standingUp &&
+      !rebuyModal &&
+      !!prevHand.distribution
+    );
+  }
+
+  return !!(
+    state === 'waiting' &&
+    myStack !== null && (myStack <= 0 || myStack < bb) &&
+    (state !== prevState || (myStack !== prevStack && prevStack > bb)) &&
```

Comment:
what is this condition about? The logic is quite hard to follow

Reply:
Check for changed hand state or changed stack. It's because we checking it in `componentWillReceiptProps`. Values can be updated at different moments of time.

But you right, it can be expressed in a more meaningful way. I'll try to refactor it.

Reply:
I'll refactor this in different PR. It's pretty old condition and wasn't added in that PR

Reply:
:+1: